// issue-manager/lib/issue.js
const mongoose = require('mongoose')

var listOfIssues = [
    {
        "id": "001",
        "title": "Learn JS",
        "status": "open",
        "assignee": "Mike",
        "created": "11-10-19",
        "updated": "11-10-19",
        "details": "Continue learning"
    },
    {
        "id": "002",
        "title": "Buy SUV",
        "status": "closed",
        "assignee": "Mike",
        "created": "04-10-19",
        "updated": "11-10-19",
        "details": "Ford Explorer"
    },
    {
        "id": "003",
        "title": "Day Job",
        "status": "open",
        "assignee": "Mike",
        "created": "01-29-18",
        "updated": "11-08-19",
        "details": "Make money"
    },
    {
        "id": "004",
        "title": "Wyatt",
        "status": "open",
        "assignee": "Chelsea",
        "created": "04-10-19",
        "updated": "11-10-19",
        "details": "Get ready"
    },
    {
        "id": "005",
        "title": "Jaxson",
        "status": "open",
        "assignee": "Chelsea",
        "created": "04-24-18",
        "updated": "11-10-19",
        "details": "Change diapers"
    }
]

const Issues = function () {
    for (var i in listOfIssues) {
        return i
        //console.log(listOfIssues[i]) 
    }
}
//res.write('\n<h1>' + JSON.stringify(listOfIssues[i]) + '</h1>');

const issueSchema = new mongoose.Schema({
    id: {
        type: String,
        required: true,
        trim: true,
    }, 
    title: {
        type: String,
        required: true,
        trim: true,
    },
    status: {
        type: String,
        required: true,
        trim: true,
    },
    assignee: {
        type: String,
        required: true,
        trim: true,
    },
    created: {
        type: String,
        required: true,
        trim: true,
    },
    updated: {
        type: String,
        required: true,
        trim: true,
    },
    details: {
        type: String,
        required: true,
        trim: true,
    }
})

const Issue = mongoose.model('Issue', issueSchema)

// const createIssue = function (id, title, status, assignee, created, updated, details) {
//     create instanceof new Issue
//     return Issue
//}

const CreateIssue = function () {
    var input = new Issue ({id, title, status, assignee, created, updated, details})
        // this.id = id
        // this.title = title,
        // this.status = status
        // this.assignee = assignee
        // this.created = created
        // this.updated = updated
        // this.details = details
    return input
}

Issue()

module.exports = {
    Issues: Issues, 
    Issue: Issue,
    CreateIssue: CreateIssue,
    listOfIssues: listOfIssues
}
